DPG/Tasks/qaK0sTrackingEfficiency.cxx

```diff
@@ -70,36 +91,39 @@ struct qaK0sTrackingEfficiency {
     registry.fill(HIST("h_EventCounter"), 1.);
 
     for (auto& v0 : fullV0s) {
-      if (v0.v0cosPA(collision.posX(), collision.posY(), collision.posZ()) > v0cospa) {
-
-        auto reconegtrack = v0.negTrack_as<PIDTracks>();
-        auto recopostrack = v0.posTrack_as<PIDTracks>();
-
-        if (TMath::Abs(v0.yK0Short()) < rapidity) {
-          if (reconegtrack.tpcNSigmaPi() < nSigTPC && recopostrack.tpcNSigmaPi() < nSigTPC) {
-            registry.fill(HIST("Test/h_R"), v0.v0radius());
-            registry.fill(HIST("Test/h_pT"), v0.pt());
-            registry.fill(HIST("Test/h_mass"), v0.mK0Short());
-
-            int negIBNhits = 0, posIBNhits = 0;
-            for (unsigned int i = 0; i < 3; i++) {
-              if (reconegtrack.itsClusterMap() & (1 << i)) {
-                negIBNhits++;
-              }
-              if (recopostrack.itsClusterMap() & (1 << i)) {
-                posIBNhits++;
-              }
-            }
-            bool negHasIB = (bool)negIBNhits;
-            bool posHasIB = (bool)posIBNhits;
-            registry.fill(HIST("Test/h_negStatus"), negHasIB);
-            registry.fill(HIST("Test/h_posStatus"), posHasIB);
-            registry.fill(HIST("Test/h_negIBhits"), negIBNhits);
-            registry.fill(HIST("Test/h_posIBhits"), posIBNhits);
-
-            registry.fill(HIST("h5_RpTmassStatus"), v0.v0radius(), v0.pt(), v0.mK0Short(), negHasIB, posHasIB);
+
+      auto reconegtrack = v0.negTrack_as<PIDTracks>();
+      auto recopostrack = v0.posTrack_as<PIDTracks>();
+
+      if (acceptV0(v0, reconegtrack, recopostrack, collision)) {
+        registry.fill(HIST("Test/h_R"), v0.v0radius());
+        registry.fill(HIST("Test/h_pT"), v0.pt());
+        registry.fill(HIST("Test/h_mass"), v0.mK0Short());
+
+        bool negHasITS = reconegtrack.hasITS();
+        bool posHasITS = reconegtrack.hasITS();
+        registry.fill(HIST("Test/h_negITSStatus"), negHasITS);
+        registry.fill(HIST("Test/h_posITSStatus"), posHasITS);
+
+        registry.fill(HIST("h5_RpTmassITSStatus"), v0.v0radius(), v0.pt(), v0.mK0Short(), negHasITS, posHasITS);
+
+        int negIBNhits = 0, posIBNhits = 0;
+        for (unsigned int i = 0; i < 3; i++) {
+          if (reconegtrack.itsClusterMap() & (1 << i)) {
+            negIBNhits++;
+          }
+          if (recopostrack.itsClusterMap() & (1 << i)) {
+            posIBNhits++;
           }
         }
+        bool negHasIB = (bool)negIBNhits;
+        bool posHasIB = (bool)posIBNhits;
+        registry.fill(HIST("Test/h_negIBStatus"), negHasIB);
+        registry.fill(HIST("Test/h_posIBStatus"), posHasIB);
+        registry.fill(HIST("Test/h_negIBhits"), negIBNhits);
+        registry.fill(HIST("Test/h_posIBhits"), posIBNhits);
+
+        registry.fill(HIST("h5_RpTmassIBStatus"), v0.v0radius(), v0.pt(), v0.mK0Short(), negHasIB, posHasIB);
```

Comment:
In the last two axes you store only a `bool` that tells you if the tracks cross the inner barrel of ITS or not and in the previous lines you just cross check the crossing in the IB. What about making everything more general and consider all the ITS layers? Also here in this spare filling you can change the last two axes with two ne ones when you store the ITS hitmap itself. What do you think? In my opinion this can make everything more flexible offline, in case of further cross checks. Let me know if it has shortcomings.